### PR TITLE
CSVGenerator Refactor

### PR DESCRIPTION
**Pull request by shaahmad**
Refactoring of CSVGenerator as promised. @breck7

### CODE REVIEW

File: charts/CSVGenerator.tsx [hunk 1]
```diff
@@ -1,91 +1,200 @@
...
+            const yearAndValue = this.yearAndValueForSingleYearDimension(
+                dim,
+                entity
+            )
+            if (yearAndValue !== null) {
+                values.push(...yearAndValue)
+                rowHasSomeValue = true
+            } else values.push("", "")
+        })
+        return rowHasSomeValue ? values : []
```
Comment (breck7) [inline charts/CSVGenerator.tsx:149]: nit: Would be a little less code and probably wouldn't be slower to ditch `rowHasSomeValue` and return something like `values.some(v => v !== "") ? values : []`

File: charts/CSVGenerator.tsx [hunk 2]
```diff
@@ -1,91 +1,200 @@
...
+            return [latestYear, latestValue]
+        } else return null
+    }
+
+    private dimensionsValues(entity: string, year: number) {
+        const values: (string | number)[] = []
+        let rowHasSomeValue = false
+
+        this.allValueDimensions.map(dim => {
+            const value = this.valueForDimensionEntityYear(dim, entity, year)
```
Comment (breck7) [inline charts/CSVGenerator.tsx:131]: nit: why the extra method and not just `dim.valueByEntityAndYear.get(entity)?.get(year)`

File: charts/CSVGenerator.tsx (hunk 1 above)
Comment (shaahmad) [reply charts/CSVGenerator.tsx]: Good idea -- removed it.

File: charts/CSVGenerator.tsx (hunk 2 above)
Comment (shaahmad) [reply charts/CSVGenerator.tsx:130]: Thought it might help with readability.

Comment (shaahmad) [reply charts/CSVGenerator.tsx:130]: Will push as is right now to enable review of other PR, but curious to hear your thoughts via Slack.

**Pull request merged**